distribution/lib/Standard/Table/0.0.0-dev/src/Extensions/Table_Ref.enso

```diff
@@ -0,0 +1,90 @@
+type Table_Ref
+    ## PRIVATE
+    Value underlying
+
+    ## PRIVATE
+       Get a column.
+       Column must implement all the expected calculations.
+       This returns a Column, but the type is not known statically because it
+       may be an in-memory or Database column.
+    at : Text | Integer -> Any ! No_Such_Column | Index_Out_Of_Bounds
+    at self selector=0 = self.underlying.at selector
+
+    ## PRIVATE
+       Resolve a Column_Ref to a Column, keeping any other values as-is.
+    resolve : Any -> Any ! No_Such_Column | Index_Out_Of_Bounds
+    resolve self value = case value of
+        Column_Ref.Name name -> self.at name
+        Column_Ref.Index index -> self.at index
+        _ -> value
+
+    ## PRIVATE
+       Resolve a Column_Ref to a Column, converting any other values into
+       a constant column.
+    resolve_as_column : Any -> Any ! No_Such_Column | Index_Out_Of_Bounds
+    resolve_as_column self value = case value of
+        Column_Ref.Name name -> self.at name
+        Column_Ref.Index index -> self.at index
+        _ -> self.underlying.make_constant_column value
+
+    ## PRIVATE
+       Transforms a condition, changing any Column_Ref instances into Column instances resolved in this table.
+    resolve_condition : Filter_Condition -> Filter_Condition
+    resolve_condition self condition = case condition of
+        Filter_Condition.Equal value -> Filter_Condition.Equal (self.resolve value)
+        Filter_Condition.Not_Equal value -> Filter_Condition.Not_Equal (self.resolve value)
+        Filter_Condition.Less value -> Filter_Condition.Less (self.resolve value)
+        Filter_Condition.Equal_Or_Less value -> Filter_Condition.Equal_Or_Less (self.resolve value)
+        Filter_Condition.Greater value -> Filter_Condition.Greater (self.resolve value)
+        Filter_Condition.Equal_Or_Greater value -> Filter_Condition.Equal_Or_Greater (self.resolve value)
+        Filter_Condition.Between lower upper -> Filter_Condition.Between (self.resolve lower) (self.resolve upper)
+        Filter_Condition.Starts_With prefix case_sensitivity -> Filter_Condition.Starts_With (self.resolve prefix) case_sensitivity
+        Filter_Condition.Ends_With prefix case_sensitivity -> Filter_Condition.Ends_With (self.resolve prefix) case_sensitivity
+        Filter_Condition.Contains prefix case_sensitivity -> Filter_Condition.Contains (self.resolve prefix) case_sensitivity
+        Filter_Condition.Not_Contains prefix case_sensitivity -> Filter_Condition.Not_Contains (self.resolve prefix) case_sensitivity
+        Filter_Condition.Like pattern -> Filter_Condition.Like (self.resolve pattern)
+        Filter_Condition.Not_Like pattern -> Filter_Condition.Not_Like (self.resolve pattern)
+        Filter_Condition.Is_In values -> Filter_Condition.Is_In (check_is_in_values "Is_In" values)
+        Filter_Condition.Not_In values -> Filter_Condition.Not_In (check_is_in_values "Not_In" values)
```

Comment:
I added this check that the `values` do not contain a `Column_Ref`, because IIRC we have agreed that there is no sense to provide a column selector widget for this case - as it can be confusing. So I guess there is neither sense to resolve the `Column_Ref`s here.

(The confusion is that normally operations work row-by-row but Is_In, when provided a column will look if the values on the left are contained _anywhere_ within the column on the right - so it's a different logic.)

So I guessed that since we do not resolve the `Column_Ref`, we should not accept it either - otehrwise it would be compared by equality and just yield to unexpected results.

So I added this errror:
![image](https://github.com/enso-org/enso/assets/1436948/3fefa59e-e0bb-4b83-9479-b1233377d530)

> Column_Ref is not allowed in Is_In to avoid unexpected behavior. As opposed to other operations, which operate on a row-by-row basis when passed a column, Is_In looks at the whole contents of the passed collection - thus usually expecting a Vector. If you want to filter the elements checking if they are present _anywhere_ in a passed column, you can pass a column by first getting it from the table using the `at` operator.

IMO it explains OK why by default this is not allowed and how the user can work around it if they know what they are doing (use `at`).

Please note that in normal flow, the user should never see this error - the widget for `Is_In` does not suggest selecting column references - it includes blank `_` nodes instead waiting for input. The error will only show up if the user _manually_ types in `Column_Ref.Name "..."` for this argument.